geopandas/io/sql.py

```diff
@@ -47,17 +47,26 @@ def read_postgis(sql, con, geom_col='geom', crs=None, hex_encoded=True,
     if geom_col not in df:
         raise ValueError("Query missing geometry column '{}'".format(geom_col))
 
-    def load_geom(x):
-        if isinstance(x, bytes):
-            return shapely.wkb.loads(x, hex=hex_encoded)
-        else:
-            return shapely.wkb.loads(str(x), hex=hex_encoded)
-    geoms = df[geom_col].apply(load_geom)
-    df[geom_col] = geoms
+    load_geom_bytes = shapely.wkb.loads
 
-    if crs is None:
-        if len(geoms) > 0:
-            srid = shapely.geos.lgeos.GEOSGetSRID(geoms[0]._geom)
+    def load_geom_buffer(x):
+        return shapely.wkb.loads(str(x))
+
+    def load_geom_str(x):
+        return shapely.wkb.loads(str(x), hex=hex_encoded)
```

Comment:
Possibly if a byte string showed up as text.
Tests pass with it hardcoded to `True`, though, and I'm not sure that ever happens.
If it's useless, then `hex_encoded` could be removed as a parameter entirely (dead code).
This might be a better question for whomever introduced `hex_encoded`, which `git blame` tells me is you: is it meant to handle bytestrings arriving as strings?

Reply:
Sorry, I should have acknowledged my complicity in my original question. :smile: Though my question was primarily driven by you having separated `load_geom_buffer` and `load_geom_str`, where only the handling of the `hex_encoded` keyword would potentially be different. I added the `hex=hex_encoded` based on my admittedly poor understanding of what it would take to make `read_postgis` compatible with spatial databases other than postGIS. Since you have more knowledge in that domain than I do, and given the way you'd separated the `load_geom` functions, I was curious if you had more insight to share. Since spatialite evidently doesn't required `hex=False` and you're not aware of any reason for that, there might not be any good reason to keep the `hex_encoded` at present.

Reply:
The `shapely.wkb.loads` `hex` parameter seems meant for Python 2, where (ASCII) text and binary can be the same type, `str`.
This doesn't happen in Python 3, where binary and text are separated as `bytes` and `str`.
Though [`sqlite3` doesn't do this][table], a Python database module could read binary into a Python 2 `str`, thus requiring the `hex` parameter to differentiate.
To summarize, in Python 2, `str` is ambiguous and may require `hex`, whereas binary-only types such as `buffer` & `bytes` and text-only types `unicode` & (Python 3) `str` are unambiguous and never need it.

If none of the database modules in the scope of this project return Python types ambiguous between binary & text, then `hex_encoded` is safe to drop.
For `sqlite3`, it is fairly safe to drop.

[table]: https://docs.python.org/2/library/sqlite3.html#introduction

Reply:
Okay, since `sqlite3` doesn't actually do that, it seems like dropping `hex_encoded` makes sense.

Reply:
Done.
Since the change is in response to a feature, I've also updated docstrings to explain the feature.